code/modules/shuttle/special.dm

```diff
@@ -258,10 +266,21 @@
 		else if(!check_times[AM] || check_times[AM] < world.time) //Let's not spam the message
 			to_chat(AM, "<span class='notice'>This ID card doesn't have an owner associated with it!</span>")
 			check_times[AM] = world.time + LUXURY_MESSAGE_COOLDOWN
-	else
+	else if(isliving(AM))
 		var/mob/living/L = AM
 		account = L.get_bank_account()
 
+	else if(isvehicle(AM))
+		var/obj/vehicle/vehicle = AM
+		for(var/passenger in vehicle.occupants)
+			if(!isliving(vehicle.occupants[passenger]))
+				continue
+			var/mob/living/rider = vehicle.occupants[passenger]
+			if(vehicle.is_driver(rider))
```

Comment:
Is it supposed to only charge the driver and allow any passengers access to the rich section for free (bearing in mind they can't leave without paying)?

Reply:
With vehicles, yeah, it's just far simpler to allow then making it break for Odysseus patients, Clown Cars, whole nine yards. Non-vehicle methods through are kept out, however.

Reply:
If it were me doing this I would want to charge every occupant or the driver for each passenger, and block access if it failed, having the gate say who couldn't pay or be paid for. This is probably saner.